### PR TITLE
This simplifies boot file and directory handling.

### PR DESCRIPTION
**Pull request by nashef**
Includes the followings:

- Adds a global verbose flag.
- Makes the repl in RBL dependent on whether the `flag-enable-repl` var is `Define`
- Simplifies the way that `--boot-dir` and `--boot` are handled.

### CODE REVIEW

File: rosette/src/BigBang.cc [hunk 1]
```diff
@@ -380,72 +380,58 @@ void InitBuiltinObs() {
...
 static FILE* FindBootFile() {
     char path[MAXPATHLEN];
-    char* RosetteLib = getenv("ROSETTE_LIB");
 
-    if (strcmp(BootFile, "") == 0) {
-        if (RosetteLib) {
-            strcpy(BootDirectory, RosetteLib);
-        }
+    strcpy(path, BootDirectory);
+    strcat(path, "/");
```
Comment (kirkwood) [inline rosette/src/BigBang.cc]: Can't be sure we're not overrunning buffer here. Suggest something like
```C++
    auto bootfile = BootFile;
    if (*bootfile == '\0') {
        bootfile = "boot.rbl";
    }

    snprintf(path, MAXPATHLEN, "%s/%s", BootDirectory, bootfile);

    // ...
```

File: rosette/rbl/rosette/async-repl.rbl [hunk 2]
```diff
@@ -128,27 +128,28 @@
...
+  	     (c-async stdin #f)
+  	     (io-catch 0 #niv)
+  	     #niv))
+  	   (else
+  	    (let [[x (if (same? expr '!!) *last-expr* expr)]]
+  	      (send eval-and-print x)
+  	      (seq
+  	       (define *last-expr* x)
+  	       (sleep)
+  	       (repl)) )))))))
```
Comment (kirkwood) [inline rosette/rbl/rosette/async-repl.rbl]: Non-lispy space.

File: rosette/src/BigBang.cc (hunk 1 above)
Comment (nashef) [reply rosette/src/BigBang.cc]: Yes, that's better. Done.

File: rosette/src/CommandLine.cc [hunk 3]
```diff
@@ -125,6 +126,18 @@ int ParseCommandLine(int argc, char** argv) {
      */
     ParanoidAboutGC = 0;
 
+    /* Set the BootDirectory default, maybe from the environment. */
+    auto boot_dir_env = secure_getenv("ROSETTE_LIB");
```
[the remark this answers is not in the document]
Comment (nashef) [reply rosette/src/CommandLine.cc]: We're only targeting Linux, atm, which is GNU.

File: rosette/src/CommandLine.cc [hunk 4]
```diff
@@ -125,6 +126,18 @@ int ParseCommandLine(int argc, char** argv) {
...
+    if (boot_dir_env) {
+        auto len = std::min(strlen(boot_dir_env), (size_t)MAXPATHLEN);
```
[the remark this answers is not in the document]
Comment (nashef) [reply rosette/src/CommandLine.cc]: Ok.

File: rosette/src/CommandLine.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (nashef) [reply rosette/src/CommandLine.cc]: I suppose there's no sense in worrying too much. Fixed.

File: rosette/src/CommandLine.cc [hunk 5]
```diff
@@ -125,6 +126,18 @@ int ParseCommandLine(int argc, char** argv) {
...
+        strncpy(BootDirectory, boot_dir_env, len);
+    } else {
+        strcpy(BootDirectory, DEFAULT);
+    }
+
+    /* Set the BootFile default. */
+    strcpy(BootFile, "boot.rbl");
```
[the remark this answers is not in the document]
Comment (nashef) [reply rosette/src/CommandLine.cc]: Ok. Done.

File: rosette/rbl/rosette/async-repl.rbl (hunk 2 above)
Comment (nashef) [reply rosette/rbl/rosette/async-repl.rbl]: Fixed.

**Pull request merged**